### PR TITLE
Added std type to logging config.

### PR DESCRIPTION
**Pull request by MFAshby**
Added demux-ing of logging to stdout/err working around
https://github.com/sirupsen/logrus/issues/403

Turned off the default logging in favour of using hooks always.

Added fallback, if no stdout/err logger was configured, add one
automatically. This prevents unexpected lack of logging after an
upgrade in case the user's configuration file is not updated.

Fixes: https://github.com/matrix-org/dendrite/issues/2054

### Pull Request Checklist

 :heavy_check_mark: 

* [x] I have added any new tests that need to pass to `sytest-whitelist` as specified in [docs/sytest.md](https://github.com/matrix-org/dendrite/blob/master/docs/sytest.md)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/master/docs/CONTRIBUTING.md#sign-off)

Signed-off-by: `Your Name <your@email.example.org>`

### CODE REVIEW

File: internal/log_unix.go [hunk 1]
```diff
@@ -17,18 +17,31 @@
...
 	"github.com/sirupsen/logrus"
 	lSyslog "github.com/sirupsen/logrus/hooks/syslog"
 )
 
-// SetupHookLogging configures the logging hooks defined in the configuration.
+// SetupLogging configures the logging hooks defined in the configuration.
 // If something fails here it means that the logging was improperly configured,
 // so we just exit with the error
-func SetupHookLogging(hooks []config.LogrusHook, componentName string) {
+func SetupLogging(hooks []config.LogrusHook, componentName string) {
```
Comment (S7evinK) [inline internal/log_unix.go:32]: Renaming this function will break building on Windows.

**Review by S7evinK — CHANGES_REQUESTED** [review]
The changes in `internal/log.go` and `setup/base/base.go` aren't necessary and like the function name will break building on Windows.

File: internal/log_unix.go (hunk 1 above)
Comment (MFAshby) [reply internal/log_unix.go:32]: Fair. Reverting this.

File: internal/log_unix.go [hunk 2]
```diff
@@ -29,6 +31,17 @@ import (
 // so we just exit with the error
 func SetupHookLogging(hooks []config.LogrusHook, componentName string) {
 	logrus.SetReportCaller(true)
+	logrus.SetFormatter(&utcFormatter{
```
Comment (S7evinK) [inline internal/log_unix.go:34]: If you move this back to `internal/log.go` and undo the changes to `setup/base/base.go`, I'm happy with this. :)

File: internal/log_unix.go [hunk 3]
```diff
@@ -49,10 +51,19 @@ func SetupHookLogging(hooks []config.LogrusHook, componentName string) {
...
 			setupSyslogHook(hook, level, componentName)
+		case "std":
+			setupStdLogHook(level)
+			stdLogAdded = true
 		default:
 			logrus.Fatalf("Unrecognised logging hook type: %s", hook.Type)
 		}
 	}
+	if !stdLogAdded {
+		logrus.Info("No std logger config found. Enabling at INFO level by default")
```
Comment (neilalexander) [inline internal/log_unix.go:62]: I'm not really sure if we need to log this? It just seems like noise.

Comment (MFAshby) [reply internal/log_unix.go:62]: Yep, I was pretty much just using it for a quick test. Could take it out.

File: internal/log_unix.go [hunk 4]
```diff
@@ -49,10 +51,19 @@ func SetupHookLogging(hooks []config.LogrusHook, componentName string) {
...
+		setupStdLogHook(logrus.InfoLevel)
+	}
+	// Hooks are now configured for stdout/err, so throw away the default logger output
+	logrus.SetOutput(ioutil.Discard)
```
[the remark this answers is not in the document]
Comment (MFAshby) [reply internal/log_unix.go:66]: I'm not sure: wouldn't this set the level for _all_ loggers?

[the remark this answers is not in the document]
Comment (neilalexander) [reply internal/log_unix.go:66]: I'm not sure to be honest. Just a thought / might be worth trying.

[the remark this answers is not in the document]
Comment (MFAshby) [reply internal/log_unix.go:66]: I can try it.

[the remark this answers is not in the document]
Comment (MFAshby) [reply internal/log_unix.go:66]: It doesn't work, I'm afraid. Calling `SetLevel(logrus.PanicLevel)` simply disables all logging 

This explains why this code here is required:  https://github.com/matrix-org/dendrite/blob/61406a6747f721e6db99fc62eeb6102ff702546f/internal/log_unix.go#L41

I think this is just a limitation of logrus.

[the remark this answers is not in the document]
Comment (neilalexander) [reply internal/log_unix.go:66]: That's unfortunate. We'll just have to go with `io.Discard` anyway then.

**Review by neilalexander — APPROVED** [review]
